src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -88,6 +89,22 @@ public void addMessage(Message message) {
     persistentStorageAgent.writeThrough(message);
   }
 
+  /** Delete a message from the current set of messages known to the application */
+  public void deleteMessage(Message message) {
+    messages.remove(message);
+    persistentStorageAgent.delete(message);
+  }
+
+  /** Delete an old message sent by this User */
+  public void deleteOldMessage(UUID userId) {
```

Comment:
It now makes sense for Messages to be stored in a Map/Table by UUID so you don't need to loop through messages every single time to find those associated with a certain user. Otherwise we could theoretically have to go through a list of millions/billions of messages (let's say we're scaling our app to the size of WhatsApp :P )